spec.md

```diff
@@ -144,18 +144,57 @@ circuit MyTop...
 
 ## Circuits
 
-All FIRRTL circuits consist of a list of modules, each representing a hardware
-block that can be instantiated. The circuit must specify the name of the
-top-level module.
+A FIRRTL circuit is a named collection of FIRRTL modules.  Each module is a
+hardware "unit" that has ports, registers, wires, and may instantiate other
+modules (see: [@sec:modules]).  (This is the same concept as a Verilog
+`module`{.verilog}.)  A public module is usable outside of the current circuit.
```

Comment:
```suggestion
`module`{.verilog}.)  A public module, when lowered, will be usable outside of the context of the current circuit.
```

Reply:
i realized upon re-reading this that we are talking about usability at the verilog level, unless we also mean usability at a hypothetical firrtl-linking level, which i think is not what we are trying to say.

Reply:
I think we're punting on FIRRTL linking and saying that this always happens at the Verilog level.  This would be analogous to a C++ header file being enough to determine the name of a function that will be compiled by another process.  There's no C++ "linking" per-se.  However, for this to work there needs to be a paired FIRRTL `extmodule` that has the same name and ports.  (Which is the same as the C++ case.)